deepmd/pt/optimizer/adamuon.py

```
@@ -0,0 +1,413 @@
+        Orthogonalized matrix in bfloat16 with same shape as input.
+
+    Raises
+    ------
+    ValueError
+        If G has fewer than 2 dimensions.
+    ValueError
+        If steps >= 100 (guard for efficiency).
+    """
+    # === Step 1. Validate ===
+    if G.ndim < 2:
+        raise ValueError("Input must have at least 2 dimensions (..., M, N).")
```

Ruff (0.14.10)

Avoid specifying long messages outside the exception class

(TRY003)